Survey/Sources/Supports/Architecture/CancelBag.swift

```diff
@@ -0,0 +1,26 @@
+//
+//  CancelBag.swift
+//  Survey
+//
+//  Created by Khanh on 25/07/2022.
+//  Copyright © 2022 Nimble. All rights reserved.
+//
+
+import Combine
+
+open class CancelBag {
```

Comment:
What is the usecase of this class? Look like current `public var subscriptions = Set<AnyCancellable>()` should already be the same as the implementation in this class. 🤔

Reply:
Agree, I think this is quite redundant

Reply:
This class is created because I want to put `Set<AnyCancelable>()` anywhere in project as a name `CancelBag` (related to `DisposeBag` in RxSwift) and it's also readable like:

> func transform(_ input: Input, cancelBag: CancelBag) -> Output 

instead of:

> func transform(_ input: Input, cancelBag: Set< AnyCancellable >) -> Output

Moreover I also recreated `store(in: ) ` method because when i pass `cancelBag` in transform function as `Set<AnyCancellable>` type then i use method `store(in: Set< AnyCancellable >)` that will produce an error because store(in: ) method cannot pass immutable value as inout argument (image below).

![Screen Shot 2022-07-26 at 14 20 17](https://user-images.githubusercontent.com/25881847/180948793-053eafe8-8034-43d5-b79d-e9c8a90fba80.png)

Does my reply answer your question correctly?
And do you have any suggestion for this one?  Thanks 🙇‍♂️ .

P/s: This idea and class i came up with from this repo https://gist.github.com/filimo/6f826b352f75cd1640e4e705340c0301

Reply:
I don't think we need to pass cancelBag from the ViewController to ViewModel. I think ViewModel can have its own cancelBag. Did I miss anything crucial that you want to pass cancelBag to ViewModel (and not having cancelBag inn ViewModel)?

If we can have cancelBag in ViewModel then we can switch back to use Set<AnyCancellable> completely.
```
func transform(_ input: Input) -> Output
```

Also if you like, you can use this to shorten your code.
```
typealias CancelBag = Set<AnyCancellable>
```

Reply:
ViewModel has its own cancelBag is fine to me. Thanks for your suggestion, I will update that change 👍 .